packages/grpc-js/src/subchannel-pool.ts

```diff
@@ -37,39 +37,81 @@ export class SubchannelPool {
     };
   } = Object.create(null);
 
+  /**
+   * A timer of a task performing a periodic subchannel cleanup.
+   */
+  private cleanupTimer: NodeJS.Timer | undefined;
```

Comment:
I slightly prefer not allowing `undefined` here, and instead having a separate boolean indicating whether the timer is running. I find it cleaner type-wise to allow as few types as possible. The downside is that the constructor needs to do `cleanupTimer=...` and `clearInterval(cleanupTimer)` and starting the timer is slightly more code, but the upside is that you never need to check the value of `cleanupTimer` itself.

Reply:
I understand that you prefer it to be more explicit, rather than having `undefined` to indicate that it's not running? I'd personally prefer to also destroy the reference to the timer if we're not running it anymore, as it ideally should not be used anymore. I thought we might be more explicit, but destroy the reference too:

```ts
type CleanupTask = { running: true; timer: Nodejs.Timer } | { running: false };
```

WDYT?

Reply:
The reason I like keeping the timer reference is it allows the code for disabling the timer to work even if the timer is already disabled, which simplifies code:

```ts
clearInterval(timer);
running = false;
```

Reply:
Hi @murgatroid99,

I couldn't find any documentation, explaining if the callback of `setInterval` is garbage collected when `clearInterval` is called on its timer. This makes me uncertain if we should keep the timer on the instance or if we should remove it. I consulted with my colleagues and no one was certain about it. So I think it might cause more confusion than clarity.

What do you think, maybe we could leave it as is?

Reply:
That's a fair point. One alternative would be to do the same thing I suggested in the constructor and replace it with a do-nothing interval and then immediately cancel it, but that kind of removes the benefit I mentioned so I'm OK with keeping it as it is here.

Reply:
OK, actually, this is a little minor, but can we have the alternate type be `null` instead of `undefined`. I dislike using `undefined` as an explicit type, and `null` would be more consistent with other code in this project. It shouldn't change much; this line can just be `private cleanupTimer: NodeJS.Timer | null = null`.

Reply:
I changed it to null and merged master, as far as I remember you made a fix to macOS builds.